### ISSUE TITLE
Cypher queries with scientific notation fail

### ISSUE THREAD

**Issue by ghost** [neo4j/neo4j#12386]
I'm using Neo4j 3.4.1 on Ubuntu.  I find that the queries 
```
CREATE (n {name:"World", freq: 3.9e-05}) RETURN "hello", n.name
CREATE (n {name:"World", freq: 3.9e05}) RETURN "hello", n.name
```
succeed _but_ 
```
CREATE (n {name:"World", freq: 3.9e+05}) RETURN "hello", n.name
```
fails with the error
```
Neo.ClientError.Statement.SyntaxError: invalid literal number (line 1, column 32 (offset: 31))
"CREATE (n {name:"World", freq: 3.9e+05}) RETURN "hello", n.name"
                                ^
```
It seems that the floating-point decimal parser does not accept a "+" preceding the exponent, but I believe that this "+" is required by the C standard (*), it is certainly implemented in this way in all C compilers that I have used, so will find its way into the formatted output of any language written in C,  so Ruby, Python, ....  I personally came across this issue using neo4j.rb, but have confirmed the issue in the bolt shell. 

(*) The standard is not available online, but IEEE Std 1003.1-2017 is, and declares itself aligned.  In their [specification for `printf`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/printf.html) it is stated
> e, E: The double argument shall be converted in the style "[-]d.ddde±dd",

**Comment by Lojjs** [comment]
@jjg-dressipi Thanks for reporting. This was a missing feature in the server-side parsing. It sounds useful, so support for this will be included in the next minor Neo4j release (Neo4j 4.1).

Best regards Louise, Neo4j Cypher team

**Comment by ghost** [comment]
Thanks! (Having fun coding against Neo4j BTW, lovely piece of work.)
